src/transformers/pipelines/__init__.py

```diff
@@ -139,10 +139,17 @@
         AutoModelForZeroShotImageClassification,
         AutoModelForZeroShotObjectDetection,
     )
+
+if is_tokenizers_available():
+    from ..tokenization_utils_fast import PreTrainedTokenizerFast
+else:
+    from ..utils.dummy_tokenizers_objects import PreTrainedTokenizerFast
```

Comment:
This import is only used for type checking, so just moving it in the type checking constant should fix all problems.

Reply:
I think I'm misunderstanding something. I tried to move it down to the section below

```python
if TYPE_CHECKING:
    from ..modeling_tf_utils import TFPreTrainedModel
    from ..modeling_utils import PreTrainedModel
    from ..tokenization_utils_fast import PreTrainedTokenizerFast
```

but that leads to the following errors

```
ERROR tests/utils/test_offline.py - RuntimeError: Failed to import transformers.pipelines because of the following error (look up to see its traceback):
name 'PreTrainedTokenizerFast' is not defined
```

as it's used in the typings in [`src/transformers/transformers/pipelines/__init__.py#L498`](https://github.com/huggingface/transformers/blob/main/src/transformers/pipelines/__init__.py#L498).

For an example of a test run with the code changes and failing tests please see #22456.

How do I do what you proposed and fix the errors?

Reply:
Got it. Thanks for clarifying! Is there an example in the code I can look at for how to do this?

Reply:
Thanks a lot for the example!